packages/server-core/src/setting/authentication-setting/authentication-setting.hooks.ts

```diff
@@ -27,21 +27,158 @@ import { hooks as schemaHooks } from '@feathersjs/schema'
 import { iff, isProvider } from 'feathers-hooks-common'
 
 import {
+  AuthenticationSettingData,
+  AuthenticationSettingType,
   authenticationSettingDataValidator,
   authenticationSettingPatchValidator,
+  authenticationSettingPath,
   authenticationSettingQueryValidator
 } from '@etherealengine/engine/src/schemas/setting/authentication-setting.schema'
+import * as k8s from '@kubernetes/client-node'
 
+import { getState } from '@etherealengine/hyperflux'
+import { BadRequest } from '@feathersjs/errors'
+import { HookContext } from '../../../declarations'
+import logger from '../../ServerLogger'
+import { ServerState } from '../../ServerState'
+import config from '../../appconfig'
 import authenticate from '../../hooks/authenticate'
 import verifyScope from '../../hooks/verify-scope'
+import { AuthenticationSettingService } from './authentication-setting.class'
 import {
   authenticationSettingDataResolver,
   authenticationSettingExternalResolver,
   authenticationSettingPatchResolver,
   authenticationSettingQueryResolver,
-  authenticationSettingResolver
+  authenticationSettingResolver,
+  authenticationSettingSchemaToDb
 } from './authentication-setting.resolvers'
 
+/**
+ * Maps settings for admin
+ * @param context
+ * @returns
+ */
+const mapSettingsAdmin = async (context: HookContext<AuthenticationSettingService>) => {
+  const auth = (Array.isArray(context.result) ? context.result : [context.result]) as AuthenticationSettingType[]
+  const loggedInUser = context.params!.user!
+  const data = auth.map((el) => {
+    if (!loggedInUser.scopes || !loggedInUser.scopes.find((scope) => scope.type === 'admin:admin'))
+      return {
+        id: el.id,
+        entity: el.entity,
+        service: el.service,
+        authStrategies: el.authStrategies,
+        createdAt: el.createdAt,
+        updatedAt: el.updatedAt
+      }
+
+    return {
+      ...el,
+      authStrategies: el.authStrategies,
+      jwtOptions: el.jwtOptions,
+      bearerToken: el.bearerToken,
+      callback: el.callback,
+      oauth: {
+        ...el.oauth
+      }
+    }
+  })
+  return context.params.paginate === false
+    ? data
+    : {
+        total: auth.length,
+        limit: context.params.query!.$limit,
+        skip: context.params.query!.$skip,
+        data
+      }
+}
+
+/**
+ * Updates OAuth in data
+ * @param context
+ * @returns
+ */
+const ensureOAuth = async (context: HookContext<AuthenticationSettingService>) => {
+  if (!context.data || context.method !== 'patch') {
+    throw new BadRequest(`${context.path} service only works for data in ${context.method}`)
+  }
+
+  const data: AuthenticationSettingData[] = Array.isArray(context.data) ? context.data : [context.data]
+  const authSettings = await context.app.service(authenticationSettingPath).get(context.id!)
+
+  if (typeof data[0].oauth === 'string') {
```

Comment:
Why is the hook only working for 0 index? It can be made generic to work for all array entries?

Reply:
Data only contains one index